lamindb/_annotate.py

```diff
@@ -349,14 +350,27 @@ class AnnDataAnnotator(DataFrameAnnotator):
 
     def __init__(
         self,
-        adata: ad.AnnData,
+        data: ad.AnnData | UPathStr,
         var_index: FieldAttr,
         categoricals: dict[str, FieldAttr] | None = None,
         using: str = "default",
         verbosity: str = "hint",
         organism: str | None = None,
     ) -> None:
-        self._adata = adata
+        from lamindb.core.storage import UPath
+
+        from ._artifact import data_is_anndata
+
+        if not data_is_anndata(data):
+            raise TypeError
+        if isinstance(data, ad.AnnData):
+            self._adata = data
+        else:
+            from lamindb.core.storage._backed_access import backed_access
+
+            self._adata = backed_access(UPath(data))
```

Comment:
Can you use `setup.core.upath.create_path()` rather than UPath, please? Only create_path incorporates access management.